controllers/account/account_controller.go

```diff
@@ -143,10 +143,28 @@ func (r *AccountReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 
 		var awsClient awsclient.Client
 		if currentAcctInstance.IsBYOC() {
-			roleToAssume := getAssumeRole(currentAcctInstance)
-			awsClient, _, err = r.assumeRole(reqLogger, currentAcctInstance, awsSetupClient, "", roleToAssume, "")
+			roleToAssume := currentAcctInstance.GetAssumeRole()
+			awsClient, _, err = handleRoleAssumption(reqLogger, r.awsClientBuilder, currentAcctInstance, r.Client, awsSetupClient, "", roleToAssume, "")
 			if err != nil {
 				reqLogger.Error(err, "failed building BYOC client from assume_role")
+				// Get custom failure reason to update account status
```

Comment:
This error handling block is duplicated 3 times in this function alone - might be worth putting it into one place, so it is easier to update it in the future.